sql/hive/src/main/scala/org/apache/spark/sql/hive/hiveUDFs.scala

```diff
@@ -59,19 +59,52 @@ private[hive] class HiveFunctionRegistry(underlying: analysis.FunctionRegistry)
 
       val functionClassName = functionInfo.getFunctionClass.getName
 
-      if (classOf[UDF].isAssignableFrom(functionInfo.getFunctionClass)) {
-        HiveSimpleUDF(new HiveFunctionWrapper(functionClassName), children)
-      } else if (classOf[GenericUDF].isAssignableFrom(functionInfo.getFunctionClass)) {
-        HiveGenericUDF(new HiveFunctionWrapper(functionClassName), children)
-      } else if (
-        classOf[AbstractGenericUDAFResolver].isAssignableFrom(functionInfo.getFunctionClass)) {
-        HiveGenericUDAF(new HiveFunctionWrapper(functionClassName), children)
-      } else if (classOf[UDAF].isAssignableFrom(functionInfo.getFunctionClass)) {
-        HiveUDAF(new HiveFunctionWrapper(functionClassName), children)
-      } else if (classOf[GenericUDTF].isAssignableFrom(functionInfo.getFunctionClass)) {
-        HiveGenericUDTF(new HiveFunctionWrapper(functionClassName), children)
-      } else {
-        sys.error(s"No handler for udf ${functionInfo.getFunctionClass}")
+      try {
+        // Based on the class of the function, we create corresponding the Hive function.
+        // To trigger any error like arguments are not valid, we call the dataType method
+        // to make sure internal Hive function resolver gets created without any issue.
+        // The only exception at here is GenericUDTF, because it does not support dataType
+        // method, we call elementTypes instead.
+        val resolvedHiveFunction =
+          if (classOf[UDF].isAssignableFrom(functionInfo.getFunctionClass)) {
+            val func = HiveSimpleUDF(new HiveFunctionWrapper(functionClassName), children)
+            // We actually only need func.dataType. Assigning the returned value to a val
+            // is just to make the compiler happy.
+            val notUsed = func.dataType
```

Comment:
hmm, it is not really safe to call `dataType` to trigger the evaluation of those lazy vals.